kitsune/users/views.py

```diff
@@ -311,6 +311,33 @@ def get(self, request):
         return super(FXAAuthenticateView, self).get(request)
 
 
+class FXAAuthenticationCallbackView(OIDCAuthenticationCallbackView):
+
+    @staticmethod
+    def get_settings(attr, *args):
+        """Override settings for Firefox Accounts.
+
+        The default values for the OIDC lib are used for the SSO login in the admin
+        interface. For Firefox Accounts we need to pass different values, pointing to the
+        correct endpoints and RP specific attributes.
+        """
+
+        val = get_oidc_fxa_setting(attr)
+        if val is not None:
+            return val
+        return super(FXAAuthenticationCallbackView,
+                     FXAAuthenticationCallbackView).get_settings(attr, *args)
+
+    def login_failure(self):
+        messages.add_message(
+            self.request,
+            messages.ERROR,
+            _('This account is not active. '
+              'Please contact an administrator if you believe this is an error')
+        )
+        return HttpResponseRedirect(reverse('home'))
+
+
 class FXALogoutView(OIDCLogoutView):
```

Comment:
Is this meant to log the user out from FxA as well?

Looking at the OIDCLogoutView it seems to expect a setting called `OIDC_OP_LOGOUT_URL_METHOD`:
https://github.com/mozilla/mozilla-django-oidc/blob/master/mozilla_django_oidc/views.py#L194

But we don't set such a setting in `get_oidc_fxa_setting`

Reply:
If this method doesn't exist it just does a logout in the django site.